test/parallel/test-tls-client-resume.js

```diff
@@ -44,32 +44,61 @@ const server = tls.Server(options, common.mustCall((socket) => {
 
 // start listening
 server.listen(0, common.mustCall(function() {
-
-  let sessionx = null;
-  let session1 = null;
+  let sessionx = null; // From right after connect, invalid for TLS1.3
+  let session1 = null; // Delivered by the session event, always valid.
+  let sessions = 0;
+  let tls13;
   const client1 = tls.connect({
     port: this.address().port,
     rejectUnauthorized: false
   }, common.mustCall(() => {
-    console.log('connect1');
+    console.log('connect1', client1.getProtocol());
+    tls13 = client1.getProtocol() === 'TLSv1.3';
     assert.strictEqual(client1.isSessionReused(), false);
     sessionx = client1.getSession();
+    assert(sessionx);
+
+    if (session1)
+      reconnect();
   }));
 
+  client1.on('data', (d) => {
```

Comment:
mustCall here?

Reply:
IIRC, whether it happens or not depends on the protocol version. I'll confirm, and document if it differs, and mustCall it if I can.

Reply:
added mustCall